docs/source/srs.rst

```diff
@@ -0,0 +1,283 @@
+
+Constraints
+-----------
+
+Below are some constraints worth considering. They are a fixed part of the
+system.
+
+* Development must be in C/C++
+* Communication will take place using CANopen
+* EVT-core will be used for low level microcontroller interfacing
+* Must be developed for the STM32F334r8
+* Resulting binary must fit within the ST microcontroller 64KB flash memory
+* Orientation measurements must be performed by the BNO055
+* non standard I2C communication with the BNO055
+
+Assumptions and Dependencies
+----------------------------
+
+It is assumed that the BNO055 chip will behave exactly as described by its
+datasheet, unless communication between the BNO055 chip and ST
+microcontroller fails.
+
+Apportioning of Requirements
+----------------------------
+
+At this point in the life cycle of the DEV1 IMU project, deployment has begun.
+Due to the nature of student-run teams, it is unlikely that there will be
+further revision to this system, as new students will likely start new projects.
+As a result, there are no future requirements planned to be added to this
+system.
+
+Specific Requirements
+=====================
+
+External Interface Requirements
+-------------------------------
+
+* no external interface?
+
+Functions
+---------
+
+The DEV1 IMU follows a specific procedure for proper initialization and
+configuration of the BNO055 chip. A general outline of what the system shall
+do is described in the Product Functions section under `Initialization
+and Configuration of BNO055`_.
+
+* should be more specific?
+
+Appendix
+========
+
+Glossary
+--------
+
+===========   ===========================================
+Term          Definition
+-----------   -------------------------------------------
+IMU           Inertial Measurement Unit
+CAN           Controller Area Network
+CANopen       Communication protocol built on CAN
+DEV1          Dirt Electric Vehicle Team
+EVT           Electrical Vehicle Team
+I2C           Inter-Integrated Circuit
+KB            Kilobyte
+STM32F334K8   ST Microcontroller selected for this project
+===========   ===========================================
+
+References
+----------
+* `BNO055 Intelligent 9-axis absolute orientation sensor <https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bno055-ds000.pdf>`_
+* `CANopen - The standardized embedded network <https://www.can-cia.org/canopen/>`_
+* `EVT-core <https://evt-core.readthedocs.io/en/latest/>`_
+* `STM32F334K8 Mainstream Mixed signals MCUs Arm Cortex-M4 core with DSP and FPU, 64 Kbytes of Flash memory, 72 MHz CPU, CCM, 12-bit ADC 5 MSPS, comparators, op-amp, hr timer <https://www.st.com/en/microcontrollers-microprocessors/stm32f334k8.html#documentation>`_
+
+Revision
+--------
+
+========    ============================          ==========
+Revision    Description                           Date
+--------    ----------------------------          ----------
+1           Initial documentation.
```

Comment:
Add a date

Reply:
Should the date be added after the documentation is finalized?

Reply:
The date should be the date you think the documentation is accurate for, so it could be pretty much any date since the last time the IMU code changed. If you want to, you can save it until right before you merge, but be sure not to forget it